scripts/_util.py

```diff
@@ -27,3 +36,33 @@ def validate_bytes(
         raise click.BadParameter("not a hex string")
 
     return decode_hex(value)
+
+
+def _account_from_keyfile(keyfile: Path, password: str) -> LocalAccount:
+    with open(keyfile, "rt") as fp:
+        privkey = Account.decrypt(json.load(fp), password)
+    return cast(LocalAccount, Account.from_key(privkey))
+
+
+def web3_for_rpc(rpc: str, account: LocalAccount) -> Web3:
+    web3 = Web3(HTTPProvider(rpc))
+
+    web3.eth.set_gas_price_strategy(rpc_gas_price_strategy)
+    web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+    web3.middleware_onion.add(construct_sign_and_send_raw_middleware(account))
+    web3.eth.default_account = account.address
+
+    return web3
```

Comment:
This function is suspiciously similar to the following from `beamer.agent`:

```python
def _make_web3(url: URL, account: LocalAccount) -> web3.Web3:
    w3 = web3.Web3(web3.HTTPProvider(url, request_kwargs=dict(timeout=5)))
    w3.eth.set_gas_price_strategy(rpc_gas_price_strategy)
    # Add POA middleware for geth POA chains, no/op for other chains
    w3.middleware_onion.inject(geth_poa_middleware, layer=0)
    w3.middleware_onion.add(construct_sign_and_send_raw_middleware(account))
    w3.eth.default_account = account.address
    return w3
```
Similarly to the `account_from_keyfile`, I would suggest to introduce a `beamer.util.make_web3` function and use it both in `beamer.agent` and under `scripts/`.